crates/nu-cli/src/commands/reduce.rs

```diff
@@ -0,0 +1,143 @@
+use crate::commands::classified::block::run_block;
+use crate::commands::WholeStreamCommand;
+use crate::prelude::*;
+use crate::{CommandArgs, CommandRegistry, Example, OutputStream};
+use futures::stream::once;
+use nu_errors::ShellError;
+use nu_protocol::{hir::Block, Primitive, Scope, Signature, SyntaxShape, UntaggedValue, Value};
+
+pub struct Reduce;
+
+#[derive(Deserialize)]
+pub struct ReduceArgs {
+    block: Block,
+    fold: Option<Value>,
+}
+
+#[async_trait]
+impl WholeStreamCommand for Reduce {
+    fn name(&self) -> &str {
+        "reduce"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("reduce")
+            .named(
+                "fold",
+                SyntaxShape::Any,
+                "reduce with initial value",
+                Some('f'),
+            )
+            .required("block", SyntaxShape::Block, "reducing function")
+    }
+
+    fn usage(&self) -> &str {
+        "Aggregate a list table to a single value using an accumulator block. Block must be
+        (A, A) -> A unless --fold is selected, in which case it may be A, B -> A."
+    }
+
+    async fn run(
+        &self,
+        args: CommandArgs,
+        registry: &CommandRegistry,
+    ) -> Result<OutputStream, ShellError> {
+        reduce(args, registry).await
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![
+            Example {
+                description: "Simple summation (equivalent to math sum)",
+                example: "echo 1 2 3 4 | reduce { = $acc + $it }",
+                result: Some(vec![UntaggedValue::int(10).into()]),
+            },
+            Example {
+                description: "Summation from starting value using fold",
+                example: "echo 1 2 3 4 | reduce -f $(= -1) { = $acc + $it }",
+                result: Some(vec![UntaggedValue::int(9).into()]),
+            },
+            Example {
+                // TODO this appears to work, but not in testing
+                description: "Folding with rows",
+                example: r#"echo a,b 1,2 3,4 | split column , | headers | reduce -f 1.6 { = $acc * $(echo $it.a | str to-int) + $(echo $it.b | str to-int) }"#,
```

Comment:
In examples, for testing purposes we only allow `echo`, math expressions, and the command being tested. You'll need to simplify this example if you want to test. Alternatively, you could use None on line 63 and turn off testing for this example.